fn/slice_test.go

```diff
@@ -241,3 +241,43 @@ func TestSliceToMap(t *testing.T) {
 		})
 	}
 }
+
+// TestHasDuplicates tests the HasDuplicates function.
+func TestHasDuplicates(t *testing.T) {
+	// Define test cases.
+	testCases := []struct {
+		name  string
+		items []int
+		want  bool
+	}{
+		{
+			name:  "All unique",
+			items: []int{1, 2, 3, 4, 5},
+			want:  false,
+		},
+		{
+			name:  "Some duplicates",
+			items: []int{1, 2, 2, 3, 4},
+			want:  true,
+		},
+		{
+			name:  "No items",
+			items: []int{},
```

Comment:
nit: there's no need to write out `[]int{}` as `nil` evaluates to an empty slice.